### DISCUSSION TITLE
DI Implementation starting the scheduler - can that be turned off?

### DISCUSSION THREAD

**Discussion by mtwoodbury** [quartznet/quartznet#1492] [Q&A] [answered]
Hello. Great tool. Thanks for making life so much easier. I do have a question in what I imagine may be a unique setup, but possibly more common in the future.

* Background:  `Monitor application`: ASP.Net with Quartz, using MS DI and the persistent store. Monitor running on a server that also has many app connection windows services that must be up at various times during the day, but `only one connection can be up at a time` for historical reasons. Running in AWS where the server can be replaced in an auto scaling group for reasons beyond our control by AWS. 
* To counter the ASG replacing a server, we have a cluster of two or more EC2 instances (Primary/Backup(s)) running the monitor service which manage between themselves who is the primary and doing the work so the connection services are started on one, but not all, and failover if the primary changes due to some failure.
* Quartz cron jobs are used to manage the communication worker service start and stop. The database is used to share the quartz tables between the server. Works well. 
 
What I have learned that during Quartz setup after the AdoJobStore.JobStoreTX does it work [startup is called](https://github.com/quartznet/quartznet/blob/main/src/Quartz/Core/QuartzScheduler.cs) by default. 

I can use the factory to get scheduler at primary/backup decision time, and put it in standby or start it, but if the Quartz setup is still in progress, it will start it. The down side now being the primary and backup(s) have a started scheduler and a backup can preempt the primary cron triggers. I do check in the Execute primary/backup and skip the job action if on a backup, but now the trigger is done.

`I could find no Quartz settings that would allow me to end up in standby mode.` Any suggestions? Anything I've missed? 
One thought was to have a special job that runs once checks the situation, and puts the scheduler in standby, then removes itself.

Do I have to stop using DI, or fork the repository to add the functionality? My reading of the Quartz clustering leads me to believe that would not help. 

Hope I'm clear. Thanks in advance.

**Comment by lahma** [comment, marked as answer]
So if you are using the MS DI integration, only thing starting the scheduler automatically would be `services.AddQuartzServer()` which adds a hosted service which starts the scheduler:

https://github.com/quartznet/quartznet/blob/3d405c0c47b97995896d5aeecca114a62c67a45a/src/Quartz.Extensions.Hosting/QuartzHostedService.cs#L28-L51

Without the hosted service you would end up with scheduler scheduler factory and you could get the scheduler from there and start it when you want to do so, following the logic in code above. Is it just this or am I missing something else?

**Reply by mtwoodbury** [reply]
Thanks very much for the answer. Didn't understand the implications of the [services.AddQuartzHostedService](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/hosted-services-integration.html#installation).  With your explanation, make sense now. Back on track now.
